nixos/modules/programs/zsh/zsh.nix

```diff
@@ -279,7 +279,7 @@ in
     environment.etc.zinputrc.source = ./zinputrc;
 
     environment.systemPackages = [ pkgs.zsh ]
-      ++ optional cfg.enableCompletion pkgs.nix-zsh-completions;
+      ++ optional (cfg.enableCompletion && ! lib.versionAtLeast (lib.getVersion config.nix.package) "2.4pre") pkgs.nix-zsh-completions;
```

Comment:
```suggestion
      ++ optional (cfg.enableCompletion && !lib.versionAtLeast (lib.getVersion config.nix.package) "2.4pre") pkgs.nix-zsh-completions;
```

Reply:
I added a commit with a pass of nixpkgs-fmt on the module.